test/functional/options/fillchars_spec.lua

```diff
@@ -0,0 +1,45 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear, execute = helpers.clear, helpers.execute
+
+describe("'fillchars'", function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(25, 5)
+    screen:attach()
+  end)
+
+  after_each(function()
+    screen:detach()
+  end)
+
+  describe('"eob" flag', function()
+    it('renders empty lines at the end of the buffer with eob', function()
+      screen:expect([[
+        ^                         |
+        ~                        |
+        ~                        |
+        ~                        |
+                                 |
+      ]])
+      execute('set fillchars+=eob:\\ ')
+      screen:expect([[
+        ^                         |
+                                 |
+                                 |
+                                 |
+        :set fillchars+=eob:\    |
+      ]])
+      execute('set fillchars+=eob:ñ')
+      screen:expect([[
+        ^                         |
+        ñ                        |
+        ñ                        |
+        ñ                        |
+        :set fillchars+=eob:ñ    |
+      ]])
```

Comment:
I would suggest also checking fullwidth (e.g. U+FF21), composing (both `a`+U+0340 and just U+0340), invalid unicode (`let &fillchars="eob:\x80"`), control characters (`let &fillchars="eob:\n"`), Unicode control characters (`let &fillchars="eob:\u0094"`), just two characters (`ab`). Note: I am not suggesting that all of this must work. Checking errors is also needed. Especially interesting just U+0340: try using `let &fillchars="diff:\u0340"` and running a diff, result is weird.